spec/template.dd

```diff
@@ -1381,7 +1381,7 @@ $(GNAME Constraint):
         The $(I Expression) is computed at compile time
         and returns a result that is converted to a boolean value.
         If that value is true, then the template is matched,
-        otherwise the template is not matched.
+        otherwise the template is not matched and compilation fails.
```

Comment:
The compilation only fails if no template matches. There might be other overloads with different constraints.